src/bridge/AdminRoomHandler.ts

```diff
@@ -420,8 +435,81 @@ export class AdminRoomHandler {
         }
     }
 
+    private async handleReconnect(req: BridgeRequest, server: IrcServer, clientList: BridgedClient[]) {
+        const client = clientList.find((c) => c.server.domain === server.domain);
+        try {
+            if (client) {
+                await client.disconnect("iwanttoreconnect", "Reconnecting", false);
+                return new MatrixAction(
+                    "notice", `Reconnecting to network...`
+                );
+            }
+            return new MatrixAction(
+                "notice", `No clients connected to this network, not reconnecting`
+            );
+        }
+        catch (err) {
+            req.log.error(err.stack);
+            return new MatrixAction(
+                "notice", `Failed to reconnect`
+            );
+        }
+    }
+
+    private async handleUsername(req: BridgeRequest, args: string[], userId: string, server: IrcServer) {
+        const domain = server.domain;
+        const store = this.ircBridge.getStore();
+        let notice;
+
+        try {
+            // Allow passwords with spaces
+            const username = args[0]?.trim();
+            if (!username || username.length === 0) {
+                notice = new MatrixAction(
+                    "notice",
+                    "Format: '!username username' " +
+                    "or '!username irc.server.name username'\n"
+                );
+            }
+            else if (username.length > IdentGenerator.MAX_USER_NAME_LENGTH) {
+                notice = new MatrixAction(
+                    "notice",
+                    `Username is longer than the maximum permitted by IRC (${IdentGenerator.MAX_USER_NAME_LENGTH}).`
+                );
+            }
+            else if (IdentGenerator.sanitiseUsername(username) !== username) {
+                notice = new MatrixAction(
+                    "notice",
+                    `Username contained invalid characters not supported by IRC.`
+                );
+            }
+            else {
+                let config = await store.getIrcClientConfig(userId, server.domain);
+                if (!config) {
+                    config = IrcClientConfig.newConfig(
+                        new MatrixUser(userId), server.domain
+                    );
+                }
+                config.setUsername(username);
+                await this.ircBridge.getStore().storeIrcClientConfig(config);
+                notice = new MatrixAction(
+                    "notice", `Successfully stored username for ${domain} for future connections.`
```

Comment:
On lines similar to this, you added a separating space.
```suggestion
                    "notice", `Successfully stored username for ${domain} for future connections. `
```